framework/utils/TreeStructure.py

```diff
@@ -53,9 +53,9 @@ def parse(inFile,dType=None):
     @ In, dType, string, optional, type of processing to use (xml or getpot)
     @ Out, tree, InputTree, structured input
   """
-  #check type of file, make sure it's a file
-  if not isinstance(inFile,file):
-    inFile = file(inFile,'r')
+  #check type of file, and open if it is a string.
+  if type(inFile).__name__ in ['str','bytes','unicode']:
+    inFile = open(inFile,'r')
```

Comment:
There is no easy way to check if a object is a file in both python3 and python2.  In python2 it is type file.  In python3 it is <class '_io.TextIOWrapper'>  I could come up with something else if needed (a function that checked which version of python we are in and did the appropriate check).